lab_2/main.py

```diff
@@ -24,16 +37,34 @@ def get_language_profiles(texts_corpus: list, language_labels: list) -> dict or
     :param language_labels: a list of given language labels
     :return: a dictionary of dictionaries - language profiles
     """
-    pass
-
+    if not isinstance(texts_corpus, list) or not isinstance(language_labels, list):
+        return None
+    for corpus in texts_corpus:
+        if not isinstance(corpus, list):
+            return None
+        for token in corpus:
+            if not isinstance(token, str):
+                return None
+    for label in language_labels:
+        if not isinstance(label, str):
+            return None
+    language_profiles = {key: get_freq_dict(value) for key, value
+                         in zip(language_labels, texts_corpus)}
```

Comment:
what happens on this line?

Reply:
Function zip creates a list of tuples where every tuple is (n-th element of language_labels, n-th element of texts_corpus). With the help of dictionary generator we create a dictionary where the certain key is the first element of the n-th tuple and his value is a frequency dictionary we get using get_freq_dict function (in this case, this function takes the second element of the n-th tuple as an argument).